src/ElasticSearch.php

```diff
@@ -131,7 +132,11 @@ public function index($documents = []): int
             ? $documents
             : new ArrayIterator($documents);
 
-        return $this->getIndexer()->buildIndex($documents);
+        try {
+            return $this->getIndexer()->buildIndex($documents);
+        } catch (ClientErrorResponseException $exception) {
+            return common_report_Report::createFailure($exception->getMessage());
```

Comment:
This method should return an integer, not a report. Why we cannot just let the exception be thrown here? 

This is a low level implementation and should not have tao report inside. It is increasing our coupling too.